timesketch/api/v1/resources/archive.py

```diff
@@ -425,53 +444,180 @@ def _export_view(self, view, sketch, zip_file):
         zip_file.writestr(f"views/{name:s}.meta", data=json.dumps(meta))
 
     def _unarchive_sketch(self, sketch: Sketch):
-        """Unarchives a sketch by opening up all indices and removing labels.
+        """Unarchives a sketch, making it and its data active again.
+
+        This method follows a transactional approach to ensure data consistency.
+        It first attempts to open all necessary OpenSearch indices associated
+        with the sketch's archived timelines. If any index fails to open for a
+        critical reason (e.g., not found), the entire operation is aborted,
+        leaving the sketch in its archived state.
+
+        If all indices are successfully opened (or confirmed to be already open),
+        the method proceeds to update the database, setting the status of the
+        sketch, its timelines, and the corresponding SearchIndex objects to 'ready'.
 
         Args:
-            sketch: Instance of timesketch.models.sketch.Sketch
+            sketch: An instance of timesketch.models.sketch.Sketch to unarchive.
+
+        Returns:
+            An integer representing the HTTP status of the operation.
+            - HTTP_STATUS_CODE_OK if successful.
+
+        Raises:
+            HTTPException:
+                - If the sketch is not currently archived.
+                - If any timeline within the sketch is in a state that prevents
+                  unarchiving (e.g., 'processing').
+                - If a critical error occurs while opening an OpenSearch index.
         """
+        logger.info("Unarchiving sketch %s ('%s').", sketch.id, sketch.name)
         if sketch.get_status.status != "archived":
             abort(
                 HTTP_STATUS_CODE_BAD_REQUEST,
                 "Unable to unarchive a sketch that wasn't already archived "
-                "(sketch: {:d})".format(sketch.id),
+                f"(sketch: {sketch.id})",
             )
 
+        # Check if any timeline is in a state that prevents unarchiving.
+        statuses_preventing_unarchival = ["processing", "fail"]
```

Comment:
Add only a warning in logs and create a Tsctl command to check for archived sketches that have failed or processing timelines in them.

Reply:
done https://github.com/google/timesketch/pull/3517